### ISSUE TITLE
[Question] Any ability to manually specify templated type subtraction?

### ISSUE THREAD

**Issue by someniatko** [vimeo/psalm#5661]
I try to type the following function of `phpoption/phpoption` library: https://psalm.dev/r/1d6b9238da

The problem is, in `Option::fromValue()` I want to specify that `V` template param is not `N`. Is it possible to do this? If not, how should I type such a function?

Thank you!

**Comment by someniatko** [comment]
The reason I want to type this, it I want to make the following work:

```php

/** @var string|null */
$stringOrNull = 'str';

$opt = Option::fromValue($stringOrNull, null);
/** @psalm-trace $opt */
```

for now `@psalm-trace` returns `Option<string|null>`, and I'd like to have `Option<string>`.

**Comment by someniatko** [comment]
BTW, I found it a bit strange that in the following case no errors are emitted:
https://psalm.dev/r/69f073e7a4

**Comment by someniatko** [comment]
> BTW, I found it a bit strange that in the following case no errors are emitted: https://psalm.dev/r/69f073e7a4

Now there are errors, great!

**Comment by niconoe-** [comment]
I'm having the same kind of issue, especially with the `map` method of the Option that takes a callable with 1st argument is of type "Same type of the Option's 1st arg minus type of the Option's 2nd arg", which is a bit tricky to handle.

Except from adding `/** @psalm-suppress InvalidArgument */` to ignore this false positive, there's seems to have no way yet to manage this.
